packages/open-truss/src/configuration/engine-v1/Frame.tsx

```diff
@@ -126,20 +141,83 @@ function processProps({
     }
   }
 
+  eachComponentSignal(componentName, COMPONENTS, (propName, signalsType) => {
+    const viewPropVal = viewProps?.[propName]
+    let signal: Signal
+    if (isSymbol(viewPropVal)) {
+      const signalName = parseSignalName(viewPropVal) ?? ''
+      signal = signals[signalName]
+    } else if (viewPropVal !== undefined) {
+      signal = signalsType.parse(viewPropVal)
+    } else {
+      signal = signals[propName]
+    }
+    addSignalToProps(propName, signal, signalsType, newProps)
+  })
+
   return {
-    data,
-    config,
     ...viewProps,
     ...newProps,
   }
 }
 
+function addSignalToProps(
+  propName: string,
+  signal: Signal<any>,
+  signalsType: SignalsZodType,
+  newProps: ComponentPropsReturnShape,
+): void {
+  const result = signalsType.safeParse(signal)
+  if (result.success) {
+    newProps[propName] = signal
+  } else {
+    throw new Error(`Signal passed to component is incorrect type.
+      Expected ${propName} to be type ${signalsType.description},
+      but got ${typeof signal}.
+    `)
+  }
```

Comment:
This is really rad, thanks for doing the renaming! I think this deserves a comment saying what this unlocks either here or up where `addSignalToProps` is used.